File: src/Python/601-700/700.SearchInBST.py

class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def searchBST(self, root, val):
        """
        :type root: TreeNode
        :type val: int
        :rtype: TreeNode
        """
        if root is None:
            return None
        if val == root.val:
            return root
        return self.searchBST(root.right, val) if val > root.val else self.searchBST(root.left, val)

if __name__ == '__main__':
    solution = Solution()
    root = TreeNode(1)
    root.left = TreeNode(2)
    root.right = TreeNode(3)
    print(solution.searchBST(root, 2))
    root = TreeNode(1)
    root.left = TreeNode(2)
    root.right = TreeNode(3)
    root.left.right = TreeNode(4)
    print(solution.searchBST(root, 4))
else:
    pass

File: src/CSharp/601-700/700.SearchInBST.cs

using System;
using System.Collections.Generic;
using System.Text;

namespace PeefyLeetCode.SearchBST
{
    public class TreeNode
    {
        public int val;
        public TreeNode left;
        public TreeNode right;
        public TreeNode(int x) { val = x; }
    }

public class Solution {
    public TreeNode SearchBST(TreeNode root, int val) {
        if (root == null)
            return null;
        if (val == root.val)
            return root;
        else if (val < root.val)
            return SearchBST(root.left, val);
        else
            return SearchBST(root.right, val);
    }
}
}
